src/DynamoCore/Engine/NodeToCode/NodeToCode.cs

```diff
@@ -320,6 +321,55 @@ public ShortestQualifiedNameReplacer(ClassTable classTable, ElementResolver reso
             this.resolver = resolver;
         }
 
+        public override AssociativeNode VisitTypedIdentifierNode(TypedIdentifierNode node)
+        {
+            if (node == null)
+                return null;
+
+            // If type is primitive type
+            if (node.datatype.UID != (int)PrimitiveType.InvalidType &&
+                node.datatype.UID < (int)PrimitiveType.MaxPrimitive)
+                return node;
+            
+            // build an idlistnode from the full type name
+            var identListNode = CoreUtils.CreateNodeFromString(node.datatype.Name);
```

Comment:
We can just pass `node.TypeAlias` to  `CreateNodeFromString`.

Reply:
will try that - thanks.